### PR TITLE
add "plan" command

### PR DESCRIPTION
**Pull request by emicklei**

### CODE REVIEW

File: commands.go [hunk 1]
```diff
@@ -60,6 +61,14 @@ func cmdCreateMigration(c *cli.Context) error {
 }
 
 func cmdMigrationsUp(c *cli.Context) error {
+	return runMigrations(c, !true)
```
Comment (jroosing) [inline commands.go]: why !true instead of false?

File: commands.go [hunk 2]
```diff
@@ -139,6 +160,17 @@ func cmdMigrationsDown(c *cli.Context) error {
 	return nil
 }
 
+func largestWithOf(list []Migration) int {
```
Comment (jroosing) [inline commands.go]: largestWidthOf, not largestWithOf

File: commands.go [hunk 3]
```diff
@@ -84,26 +93,38 @@ func cmdMigrationsUp(c *cli.Context) error {
...
+		if isLogOnly {
+			leadingTitle = execPlan
 		}
-		mtx.lastApplied = each.Filename
-		// save after each succesful migration
-		if err := mtx.stateProvider.SaveState(mtx.lastApplied); err != nil {
-			reportError(mtx.stateProvider.Config(), "save state", err)
-			return errAbort
+		log.Printf("%s %-"+strconv.Itoa(prettyWidth)+"s (%s)\n", leadingTitle, pretty(each.Filename), each.Filename)
+		if isLogOnly {
```
Comment (jroosing) [inline commands.go]: Extract to a func that accepts 1 param (isLogOnly) 

will reduce the size of this function significantly maybe a runAll private function?

File: commands.go [hunk 4]
```diff
@@ -152,13 +184,7 @@ func cmdMigrationsStatus(c *cli.Context) error {
...
 	log.Println(statusSeparator)
 	var last string
-	prettyWidth := 0
-	for _, each := range all {
-		pf := pretty(each.Filename)
-		if len(pf) > prettyWidth {
-			prettyWidth = len(pf)
-		}
-	}
+	prettyWidth := largestWithOf(all)
```
Comment (jroosing) [inline commands.go]: Don't forget to rename this after renaming the function

File: commands.go [hunk 5]
```diff
@@ -84,26 +93,38 @@ func cmdMigrationsUp(c *cli.Context) error {
 		reportError(mtx.stateProvider.Config(), "up until stop", errors.New("No such migration file: "+stopAfter))
 		return errAbort
 	}
+	prettyWidth := largestWithOf(all)
```
Comment (jroosing) [inline commands.go]: largestWidthOf
note the D

File: commands.go (hunk 1 above)
Comment (emicklei) [reply commands.go]: looks nice

File: commands.go (hunk 2 above)
Comment (emicklei) [reply commands.go]: typo

File: commands.go (hunk 4 above)
Comment (emicklei) [reply commands.go]: refactor tooling helps here

File: commands.go (hunk 1 above)
Comment (jroosing) [reply commands.go]: :' )

**Pull request merged**